### PR TITLE
add persistance policy for platform components

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

Adds persistance policy support for below components
alertmanager
prometheus
elasticsearch
postgresql
stan

scope: for POC adding these policy cleans up the pvc after the uninstallation of platform

## Related Issues

https://github.com/astronomer/astronomer/issues/1974

## Merging

cherry-pick to release-0.35 and release-0.36

### CODE REVIEW

File: charts/elasticsearch/templates/data/es-data-statefulset.yaml [hunk 1]
```diff
@@ -13,6 +13,9 @@ metadata:
     heritage: {{ .Release.Service }}
     role: data
 spec:
+  {{- if and .Values.common.persistence.enabled .Values.common.persistence.persistentVolumeClaimRetentionPolicy }}
+  persistentVolumeClaimRetentionPolicy: {{- toYaml .Values.common.persistence.persistentVolumeClaimRetentionPolicy  | nindent 4 }}
```
Comment (danielhoherd) [inline charts/elasticsearch/templates/data/es-data-statefulset.yaml:17]:
```suggestion
  persistentVolumeClaimRetentionPolicy: {{- toYaml .Values.common.persistence.persistentVolumeClaimRetentionPolicy | nindent 4 }}
```

File: charts/elasticsearch/templates/master/es-master-statefulset.yaml [hunk 2]
```diff
@@ -13,6 +13,9 @@ metadata:
     heritage: "{{ .Release.Service }}"
     role: master
 spec:
+  {{- if and .Values.common.persistence.enabled .Values.common.persistence.persistentVolumeClaimRetentionPolicy }}
+  persistentVolumeClaimRetentionPolicy: {{- toYaml .Values.common.persistence.persistentVolumeClaimRetentionPolicy  | nindent 4 }}
```
Comment (danielhoherd) [inline charts/elasticsearch/templates/master/es-master-statefulset.yaml:17]:
```suggestion
  persistentVolumeClaimRetentionPolicy: {{- toYaml .Values.common.persistence.persistentVolumeClaimRetentionPolicy | nindent 4 }}
```

File: charts/postgresql/templates/statefulset-slaves.yaml [hunk 3]
```diff
@@ -9,6 +9,9 @@ metadata:
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
 spec:
+  {{- if and .Values.persistence.enabled .Values.persistence.persistentVolumeClaimRetentionPolicy }}
+  persistentVolumeClaimRetentionPolicy: {{- toYaml .Values.persistence.persistentVolumeClaimRetentionPolicy  | nindent 4 }}
```
Comment (danielhoherd) [inline charts/postgresql/templates/statefulset-slaves.yaml:13]:
```suggestion
  persistentVolumeClaimRetentionPolicy: {{- toYaml .Values.persistence.persistentVolumeClaimRetentionPolicy | nindent 4 }}
```

File: charts/postgresql/templates/statefulset.yaml [hunk 4]
```diff
@@ -12,6 +12,9 @@ metadata:
     argocd.argoproj.io/sync-wave: "-1"
   {{- end }}
 spec:
+  {{- if and .Values.persistence.enabled .Values.persistence.persistentVolumeClaimRetentionPolicy }}
+  persistentVolumeClaimRetentionPolicy: {{- toYaml .Values.persistence.persistentVolumeClaimRetentionPolicy  | nindent 4 }}
```
Comment (danielhoherd) [inline charts/postgresql/templates/statefulset.yaml:16]:
```suggestion
  persistentVolumeClaimRetentionPolicy: {{- toYaml .Values.persistence.persistentVolumeClaimRetentionPolicy | nindent 4 }}
```

File: charts/prometheus/templates/prometheus-statefulset.yaml [hunk 5]
```diff
@@ -12,6 +12,9 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 spec:
+  {{- if and .Values.persistence.enabled .Values.persistence.persistentVolumeClaimRetentionPolicy }}
+  persistentVolumeClaimRetentionPolicy: {{- toYaml .Values.persistence.persistentVolumeClaimRetentionPolicy  | nindent 4 }}
```
Comment (danielhoherd) [inline charts/prometheus/templates/prometheus-statefulset.yaml:16]:
```suggestion
  persistentVolumeClaimRetentionPolicy: {{- toYaml .Values.persistence.persistentVolumeClaimRetentionPolicy | nindent 4 }}
```

File: charts/stan/templates/statefulset.yaml [hunk 6]
```diff
@@ -15,6 +15,9 @@ metadata:
     argocd.argoproj.io/sync-wave: "-1"
   {{- end }}
 spec:
+  {{- if and .Values.store.volume.enabled  .Values.persistence.persistentVolumeClaimRetentionPolicy }}
+  persistentVolumeClaimRetentionPolicy: {{- toYaml .Values.persistence.persistentVolumeClaimRetentionPolicy  | nindent 4 }}
```
Comment (danielhoherd) [inline charts/stan/templates/statefulset.yaml:19]:
```suggestion
  persistentVolumeClaimRetentionPolicy: {{- toYaml .Values.persistence.persistentVolumeClaimRetentionPolicy | nindent 4 }}
```

**Comment by pgvishnuram** [comment]
@danielhoherd see if this is mergeable

File: tests/chart_tests/test_prometheus_statefulset.py [hunk 7]
```diff
@@ -127,3 +127,26 @@ def test_prometheus_with_multiple_extraFlags(self, kube_version):
             "--enable-feature=remote-write-receiver" in c_by_name["prometheus"]["args"]
         )
         assert "--enable-feature=agent" in c_by_name["prometheus"]["args"]
+
```
Comment (danielhoherd) [inline tests/chart_tests/test_prometheus_statefulset.py:130]: Prometheus is missing the `"persistentVolumeClaimRetentionPolicy" not in doc["spec"]` test case

File: tests/chart_tests/test_elasticsearch.py [hunk 8]
```diff
@@ -650,3 +650,34 @@ def test_elasticsearch_nginx_deployment_overrides(self, kube_version):
         assert len(docs) == 1
         c_by_name = get_containers_by_name(docs[0])
         assert c_by_name["nginx"]["securityContext"] == {"runAsNonRoot": True}
+
```
Comment (danielhoherd) [inline tests/chart_tests/test_elasticsearch.py:653]: Elasticsearch is missing the `"persistentVolumeClaimRetentionPolicy" not in doc["spec"]` test case.

File: tests/chart_tests/test_postgres.py [hunk 9]
```diff
@@ -70,3 +70,64 @@ def test_postgresql_statefulset_with_private_registry_enabled(self, kube_version
                 assert container["image"].startswith(
                     repository
                 ), f"Container named '{name}' does not use registry '{repository}': {container}"
+
```
Comment (danielhoherd) [inline tests/chart_tests/test_postgres.py:73]: Postgres is missing the `"persistentVolumeClaimRetentionPolicy" not in doc["spec"]` test case

**Review by danielhoherd — APPROVED** [review]
Looks mostly good. Some test cases are missing, which I've commented. We can add those if they make sense, but otherwise this is good to merge.

**Pull request merged**